### PR TITLE
[primTorch] Add refs for `softmax`, `softmin`, `log_softmax`

### PR DESCRIPTION
**Pull request by nkaretnikov**
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #84957
* __->__ #84956

cc @ezyang @mruberry @ngimel @Lezcano @fdrocha

### CODE REVIEW

**Review by lezcano — APPROVED** [review]
Cool

File: torch/_refs/nn/functional/__init__.py [hunk 1]
```diff
@@ -233,6 +236,36 @@ def selu(a: TensorLikeType, inplace: bool = False) -> TensorLikeType:
...
+    dim: Optional[int] = None,
+    _stacklevel: int = 3,  # for compat when using TorchRefsMode(strict=True)
+    dtype: Optional[torch.dtype] = None,
+) -> TensorLikeType:
+    # The error is for compat with regular PyTorch, which has this behavior
+    # deprecated.  For PrimTorch, it's fine to drop support for deprecated
+    # behavior because it requires explicit opt in.  This error is to inform
+    # users how to update their calls.
+    check(dim is not None, lambda: "implicit dim not supported, use dim=X")
+    return torch.softmax(a=a, dim=dim, dtype=dtype)  # type: ignore[call-overload]
```
Comment (mruberry) [inline torch/_refs/nn/functional/__init__.py:251]: This reference basically assumes a softmax prim -- but should softmax actually be a prim? 

If we want to do this to unblock something we should file an issue and comment here that this isn't actually going to a prim, and the hard work of implementing softmax remains.

File: torch/_refs/nn/functional/__init__.py [hunk 2]
```diff
@@ -357,6 +390,21 @@ def l1_loss(
...
+    dim: Optional[int] = None,
+    _stacklevel: int = 3,  # for compat when using TorchRefsMode(strict=True)
+    dtype: Optional[torch.dtype] = None,
+) -> TensorLikeType:
+    # The error is for compat with regular PyTorch, which has this behavior
+    # deprecated.  For PrimTorch, it's fine to drop support for deprecated
+    # behavior because it requires explicit opt in.  This error is to inform
+    # users how to update their calls.
+    check(dim is not None, lambda: "implicit dim not supported, use dim=X")
+    return torch.log_softmax(a=a, dim=dim, dtype=dtype)  # type: ignore[call-overload]
```
Comment (mruberry) [inline torch/_refs/nn/functional/__init__.py:405]: Same concern about `log_softmax` -- do we expect it to be prim? I don't think so. Shouldn't these references implement the algorithms in a numerically accurate way?

File: torch/_refs/special/__init__.py [hunk 3]
```diff
@@ -60,6 +61,24 @@ def logit(self: TensorLikeType, eps: Optional[float] = None) -> TensorLikeType:
     return torch.log(torch.true_divide(self, torch.sub(1, self)))
 
 
+# CompositeImplicitAutograd - don't register decomp
```
Comment (mruberry) [inline torch/_refs/special/__init__.py:64]: Nice forwarding aliases -- a brief comment to that effect would be nice but nbd

**Review by mruberry — COMMENTED** [review]
Edit: I didn't see the refs in the `torch` namespace -- my bad, this looks fine.

File: torch/_refs/nn/functional/__init__.py (hunk 2 above)
Comment (Lezcano) [reply torch/_refs/nn/functional/__init__.py:405]: The refs are implemented in the `torch.` namespace. These refs deal with those pesky `_stacklevel` args. See https://github.com/pytorch/pytorch/pull/84502#discussion_r963578734

File: torch/_refs/nn/functional/__init__.py (hunk 1 above)
Comment (ngimel) [reply torch/_refs/nn/functional/__init__.py:251]: This reference doesn't assume softmax prim, it is just an alias.

Comment (mruberry) [reply torch/_refs/nn/functional/__init__.py:251]: Aha! Thank you for the clarification

**Review by mruberry — APPROVED** [review]
Cool!

**Comment by ZainRizvi** [comment]
@pytorchbot revert -m "This is causing the MPS test test_output_match_log_softmax_with_dtype_cpu_float32 (__main__.TestConsistencyCPU) to fail" -c 'landrace'

Example logs:
https://hud.pytorch.org/pytorch/pytorch/commit/c08c7997503fbe8472a957f712322d5fb5fa11bf

**Comment by nkaretnikov** [comment]
@kulinseth I changed `variant_test_name` in `log_softmax`'s OpInfo to be consistent with other ops, so I had to update the skip in the MPS tests.
